lib/pos_blockchain.go

```diff
@@ -472,6 +488,48 @@ func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 	return utxoView, nil
 }
 
+// BuildUtxoViewToBlockParent builds a UtxoView to the block provided. It does this by
+// identifying all uncommitted ancestors of this block and then connecting those blocks.
+func (bc *Blockchain) BuildUtxoViewToBlockParent(parentHash BlockHash) (*UtxoView, error) {
+	uncommittedAncestors := []*BlockNode{}
+	currentBlock := bc.blockIndex[parentHash]
+	if currentBlock == nil {
+		return nil, errors.Errorf("BuildUtxoViewToBlockParent: Block %v not found in block index", parentHash)
+	}
+	for currentBlock.CommittedStatus == UNCOMMITTED {
```

Comment:
What if the input block is committed but it is not the committed tip? We wouldn't be able to build a UtxoView for it. It'll be safest to perform that check here and error in that case. This is definitely one function I think is worth bullet proofing, by assuming that the caller may unknowingly pass in a bad input